### PR TITLE
feat(custom watermark)

### PR DESCRIPTION
**Pull request by Mr-LLLLL**
#93

### CODE REVIEW

**Comment by Mr-LLLLL** [comment]
maybe custom watermark by use lua function is good than set to `cwd`

File: README.md [hunk 1]
```diff
@@ -333,7 +333,7 @@ There is a default config:
     title = "CodeSnap.nvim",
     code_font_family = "CaskaydiaCove Nerd Font",
     watermark_font_family = "Pacifico",
-    watermark = "CodeSnap.nvim",
+    watermark = "CodeSnap.nvim" -- or function return "CodeSnap.nvim" end,
```
Comment (mistricky) [inline README.md:336]: There is no need to add comments, cause this chapter just shows what the default values of config is

File: lua/codesnap/config.lua [hunk 2]
```diff
@@ -46,6 +46,10 @@ function config_module.get_config(extension)
   }, static.config)
 
   config.save_path = parse_save_path(config.save_path)
+  if type(config.watermark) == "function" then
+    local mark = config.watermark()
+    config.watermark = type(mark) == "string" and mark or ""
+  end
```
Comment (mistricky) [inline lua/codesnap/config.lua:52]: The `get_config` function will called every time when users trigger the take snapshot command, the watermark logic may not be appropriate here, I tend to place this logic under the `setup` function, you can refer to [here](https://github.com/mistricky/codesnap.nvim/blob/main/lua/codesnap/init.lua?plain=1#L11-L13). The setup function will only be called once when the user invokes the setup function.

**Comment by mistricky** [comment]
Hi @Mr-LLLLL, thx for this contribution. Yes, allowing users to pass a function to dynamically calculate what watermark should display is a really useful feature. 

However, the design of the watermark allows users to add custom symbols or something on their screenshots, thus it will make screenshots more personalized. I will merge this PR but the feature request mentioned on https://github.com/mistricky/codesnap.nvim/issues/93 has not be resolved yet, maybe I will open another PR to implement it. Thx for this contribution, and thx for using CodeSnap ❤️

File: README.md (hunk 1 above)
Comment (Mr-LLLLL) [reply README.md:336]: all right

**Comment by Mr-LLLLL** [comment]
> Hi @Mr-LLLLL, thx for this contribution. Yes, allowing users to pass a function to dynamically calculate what watermark should display is a really useful feature.
> 
> However, the design of the watermark allows users to add custom symbols or something on their screenshots, thus it will make screenshots more personalized. I will merge this PR but the feature request mentioned on #93 has not be resolved yet, maybe I will open another PR to implement it. Thx for this contribution, and thx for using CodeSnap ❤️

I think your talking is right. watermark is not good way to show `workspace` but keep this feature is good idea 'custom watermark by lua function'

**Comment by mistricky** [comment]
Wait, I just realized your watermark function is called without any arguments, it's unnecessary to change the static value to function way, because the setup function itself is a computational environment:

```lua
setup({
  ..
  watermark: "foo" + "bar" + "any value"
})
```

**Comment by mistricky** [comment]
I will close this PR, because the function invoke way is unnecessary, thx for your contribution

**Comment by Mr-LLLLL** [comment]
ok

**Comment by Mr-LLLLL** [comment]
:)

**Pull request closed without merging**